lib/puppet/provider/cpdb_create/podman.rb

```diff
@@ -0,0 +1,42 @@
+Puppet::Type.type(:cpdb_create).provide(:podman) do
+
+  commands :podman => '/bin/podman'
+
+  def create
+    create_database
+    write_done_file
+  end
+
+  def exists?
+    File.exist?(done_file)
+  end
+
+  private
+
+  def create_database
+    podman(
```

Comment:
Do quadlets provide a way to reuse configuration from the `.container` file and execute a command within the right context? If not, perhaps we should work with the developers to get such a feature.

Mostly because I'd be worried about missing volumes that `cpdb` may use.

Reply:
I'm not following what you are asking here.

Reply:
Essentially, is there something like `quadlet-exec candlepin.container /usr/share/cpdb ...` that reads `candlepin.container` for all the config options (network, container image, volumes, etc) and does the right podman invocation for you?

Reply:
I will look around and see what I can find. Typically from what I have seen in the container world, you build into the container start-up checking for and performing any migrations that are needed. In fact, Candlepin 4.4 has switched to this model to handle migrations at start up of the process. That model would remove the need for this all together in the container world.

Reply:
Would it make sense to first move to 4.4 then?

Looking further, we may still need a similar thing for foreman-rake unless we also replace all functionality we now have. Perhaps not a bad thing either: make rake developer/builder only and take it away from users

Reply:
I did some initial testing of 4.4, and with their new configuration option:

```
candlepin.db.database_manage_on_startup = None|Report|Halt|Manage

None = The database will not be checked and no updates will be made.
Report = Either a confirmation that the db is up to date or the names of the changesets that have not been run. Information sent to the log either at info level or warn level based on the result.
Halt = Same as for Report, but a RuntimeException will be thrown to stop the Candlepin startup if there are unrun changesets.
Manage = Same as for Report, but a Liquibase update will attempt to bring the database schema up to the correct version. An unsuccessful update will result in the throwing of a RuntimeException.
```

Using `Manage` neither cp_create or cp_update is needed which would eliminate the providers for both standard and container based deployments. This also solves any issues with cpdb and passwords as it reads it from the config file rather than as inputs.

I added a commit to this PR that uses a 4.4 image I built to demonstrate that.